From: Petr Tesarik <ptesarik@suse.com>
Date: Fri, 4 May 2018 12:55:44 +0200
Subject: Block initrd-parse-etc.service until dump is saved
References: bsc#1091186
Upstream: merged
Git-commit: ac14396b33a7d507a65999aa7f70668f9cd15e8e

During boot, custom services like kdump-save.service run in parallel
with other generic initrd services up until initrd.target. However,
a reload from real root may change the dependency graph in a way
that prevents kdump-save.service from finishing. Since kdump needs
only the explicitly listed mount points, boot process may pause
while the dump is being saved. In fact, it is preferable to do so,
lest other processes interfere with kdump itself in the restricted
environment.

Signed-off-by: Petr Tesarik <ptesarik@suse.com>
---
 init/kdump-save.service.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/init/kdump-save.service.in b/init/kdump-save.service.in
index 7879dc2..a4d6a65 100644
--- a/init/kdump-save.service.in
+++ b/init/kdump-save.service.in
@@ -11,7 +11,7 @@
 Description=save kernel crash dump
 Documentation=man:kdump(7)
 RequiresMountsFor=@KDUMP_MOUNTPOINTS@
-Before=initrd-cleanup.service
+Before=initrd-parse-etc.service
 ConditionPathExists=/etc/initrd-release
 ConditionPathExists=/proc/vmcore
 
-- 
2.13.6

